receiver/kubeletstatsreceiver/internal/kubelet/metadata.go

```diff
@@ -50,15 +51,98 @@ type Metadata struct {
 	Labels                    map[MetadataLabel]bool
 	PodsMetadata              *v1.PodList
 	DetailedPVCResourceSetter func(rb *metadata.ResourceBuilder, volCacheID, volumeClaim, namespace string) error
+	PodResources              map[string]resources
+	ContainerResources        map[string]resources
+}
+
+type resources struct {
+	cpuLimit      float64
+	cpuRequest    float64
+	memoryLimit   float64
+	memoryRequest float64
+}
+
+func getContainerResources(r *v1.ResourceRequirements) resources {
+	if r == nil {
+		return resources{}
+	}
+	var containerResource resources
+	cpuLimit, err := strconv.ParseFloat(r.Limits.Cpu().AsDec().String(), 64)
+	if err == nil {
+		containerResource.cpuLimit = cpuLimit
+	}
+	cpuRequest, err := strconv.ParseFloat(r.Requests.Cpu().AsDec().String(), 64)
```

Comment:
For CPU we can use `MilliValue` or `ScaledValue` instead parsing string

Reply:
I think we'll want to keep CPU metrics as a float so that we can represent decimal values.  The unit for the CPU metrics should be `{core}` to match Kube State Metrics, so we need to represent `100m` as `0.1`.  The `AsDec()` solution handled this (and then was carried over to Memory incorrectly, you're right that those can just be int4), but we can simplify it with `AsApproximateFloat64`.